src/yoti_request/aml.result.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+// AML result attributes
+const ON_PEP_LIST_ATTR = 'on_pep_list';
+const ON_FRAUD_LIST_ATTR = 'on_fraud_list';
+const ON_WATCH_LIST_ATTR = 'on_watch_list';
+
+exports.AmlResult = class AmlResult {
+
+  /**
+   * Check if all expected attributes are included in the result.
+   *
+   * @param rawResult
+   */
+  static checkAttributes (rawResult) {
+    if (!rawResult instanceof Array) {
+      throw new Error('Result Data should be an array');
+    }
+
+    let expectedElements = [ON_PEP_LIST_ATTR, ON_FRAUD_LIST_ATTR, ON_WATCH_LIST_ATTR];
+
+    for (let key in expectedElements) {
```

Comment:
Never use `for in` to loop through Arrays in JavaScript. You can replace it with something like `for (let i = 0; i < expectedElements.length; i++) {`

Reply:
Never? the difference I know is `for ...in` syntax only returns index of elements that have value in it. If there are other reasons not to use it, it will be good to share them.

Reply:
Yes, never, because it's not meant to do that. It iterates over the enumerable properties of an object. https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea

Reply:
I could change that, although it wouldn't have affected the current implementation since I am using `if (!rawResult.hasOwnProperty(attr)) {`.

Reply:
The code works, as long as somewhere in the code or a library a property is not being added to an object. Then it breaks.